Bug: https://bugs.launchpad.net/ubuntu/+source/libvirt/+bug/1418221
Author: Scott Moser <smoser@ubuntu.com>
Description: get libvirt functional with qemu on ppc64le 
 This is just a hack, but seems functional.  We essentially say that
 wherever libvirt sees 'ppc64le' in uname machine, it should just act
 like it had seen a ppc64.  That is desireable as ppc64le as a qemu
 host is basically identical to ppc64(big endian).
 .
 One thing to note, is that we have to use cpu_mode=host-passthrough
 /etc/nova/nova-compute.conf as the default value 'host-model' will
 render to a qemu-cmdline that includes '-cpu host,(null)'
 .
 Note also that src/qemu/qemu_capabilities.c:virQEMUCapsInit might 
 seem like a better place for this hack, but my recollection is that
 doing the change there results in 'ppc' rather than 'ppc64' as the
 default system (ie, using qemu-system-ppc).
 .
 It also almost certainly would break libvirt-lxc on a ppc64le host.
--- a/src/util/virarch.c
+++ b/src/util/virarch.c
@@ -169,6 +169,8 @@ virArch virArchFromHost(void)
         arch = VIR_ARCH_I686;
     } else if (STREQ(ut.machine, "amd64")) {
         arch = VIR_ARCH_X86_64;
+    } else if (STREQ(ut.machine, "ppc64le")) {
+        arch = VIR_ARCH_PPC64;
     } else {
         /* Otherwise assume the canonical name */
         if ((arch = virArchFromString(ut.machine)) == VIR_ARCH_NONE) {
